web/client-api/src/main/java/io/deephaven/web/client/api/barrage/BarrageUtils.java

```diff
@@ -182,54 +177,46 @@ private static ColumnData readArrowBuffer(ByteBuffer data, Iter<BarrageFieldNode
+                assert positions.length().toFloat64() >= size;
+                Int8Array byteArray = new Int8Array(TypedArrayHelper.unwrap(data).buffer, (int) positions.offset().toFloat64(), size);
                 return new ByteArrayColumnData(Js.uncheckedCast(byteArray));
             case "double":
-                assert positions.length().toFloat64() - positions.offset().toFloat64() >= size * 8;
-                Float64Array doubleArray = new Float64Array(TypedArrayHelper.unwrap(data).buffer, (int) (data.position() + positions.offset().toFloat64()), size);
-                data.position((int) (data.position() + positions.length().toFloat64() - positions.offset().toFloat64()));
+                assert positions.length().toFloat64() >= size * 8;
+                Float64Array doubleArray = new Float64Array(TypedArrayHelper.unwrap(data).buffer, (int) positions.offset().toFloat64(), size);
                 return new DoubleArrayColumnData(Js.uncheckedCast(doubleArray));
             case "float":
-                Float32Array floatArray = new Float32Array(TypedArrayHelper.unwrap(data).buffer, (int) (data.position() + positions.offset().toFloat64()), size);
-                assert positions.length().toFloat64() - positions.offset().toFloat64() >= size * 4;
-                data.position((int) (data.position() + positions.length().toFloat64() - positions.offset().toFloat64()));
+                assert positions.length().toFloat64() >= size * 4;
+                Float32Array floatArray = new Float32Array(TypedArrayHelper.unwrap(data).buffer, (int) positions.offset().toFloat64(), size);
                 return new FloatArrayColumnData(Js.uncheckedCast(floatArray));
             case "char":
-                assert positions.length().toFloat64() - positions.offset().toFloat64() >= size * 2;
-                Uint16Array charArray = new Uint16Array(TypedArrayHelper.unwrap(data).buffer, (int) (data.position() + positions.offset().toFloat64()), size);
-                data.position((int) (data.position() + positions.length().toFloat64() - positions.offset().toFloat64()));
+                assert positions.length().toFloat64() >= size * 2;
+                Uint16Array charArray = new Uint16Array(TypedArrayHelper.unwrap(data).buffer, (int) positions.offset().toFloat64(), size);
                 return new CharArrayColumnData(Js.uncheckedCast(charArray));
             // longs are a special case despite being java primitives
             case "long":
             case "io.deephaven.db.tables.utils.DBDateTime":
-                assert positions.length().toFloat64() - positions.offset().toFloat64() >= size * 8;
+                assert positions.length().toFloat64() >= size * 8;
                 long[] longArray = new long[size];
 
-                int startPos = data.position();
-                data.position((int) (startPos + positions.offset().toFloat64()));
+                data.position((int) positions.offset().toFloat64());
```

Comment:
Why do we need to do this for date-time, etc, but not for primitives? Are we mixing positional and absolute access, and are we doing that on purpose?

Reply:
JS has "typed arrays" for other primitives, contiguous blocks of memory we can address by position, with fixed size values. There is no such typed array for int64, because js can't accurately represent int64 in its "number" type, so we read these values the hard way.

I would be okay with changing this method to only use absolute position in the buffer (just `positions.offset().toFloat64() + i`), but we already need to mix absolute and relative for things like ByteBuffer.get(byte[]), since there is no variant that takes absolute position, so I didn't see an issue with doing the same here.